src/ps/signature.ts

```diff
@@ -0,0 +1,366 @@
+   */
+  verify(messages: Uint8Array[], publicKey: PSPublicKey, params: PSSignatureParams): VerifyResult {
+    if (messages.length !== params.supportedMessageCount()) {
+      throw new Error(
+        `Number of messages ${
+          messages.length
+        } is different from ${params.supportedMessageCount()} supported by the signature params`
+      );
+    }
+    if (messages.length !== publicKey.supportedMessageCount()) {
+      throw new Error(
+        `Number of messages ${
+          messages.length
+        } is different from ${publicKey.supportedMessageCount()} supported by the public key`
+      );
+    }
+    return psVerify(messages, this.value, publicKey.value, params.value);
+  }
+
+  /**
+   * Aggregates signatures received from participants.
+   * @param signatures
+   * @param h
+   * @returns
+   */
+  static aggregate(signatures: Map<number, PSSignature>, h: Uint8Array): PSSignature {
+    const rawSignatures = new Map([...signatures.entries()].map(([participant, sig]) => [participant, sig.value]));
+
+    return new PSSignature(psAggregateSignatures(rawSignatures, h));
+  }
+}
+
+/**
+ * Modified Pointcheval-Sanders blind signature used in `Coconut`.
+ */
+export class PSBlindSignature extends BytearrayWrapper {
+  /**
+   * Generate blinding for creating the commitment used in the request for blind signature
+   * @param seed - Optional seed to serve as entropy for the blinding.
+   */
+  static generateBlinding(seed?: Uint8Array): Uint8Array {
+    return generateRandomFieldElement(seed);
+  }
+
+  /**
+   * Generates a blind signature over the commitment of unknown messages and known messages
+   * @param commitment - Commitment over unknown messages sent by the requester of the blind signature. Its assumed that
+   * the signers has verified the knowledge of committed messages
+   * @param knownMessages
+   * @param secretKey
+   * @param params
+   */
+  static generate(messages: Iterable<PSCommitmentOrMessage>, secretKey: PSSecretKey, h: Uint8Array): PSBlindSignature {
+    return new PSBlindSignature(psBlindSign(messages, secretKey.value, h));
+  }
+
+  /**
+   * Unblind the blind signature to get a regular signature that can be verified
+   * @param blinding
+   */
+  unblind(indexedBlindings: Map<number, Uint8Array>, pk: PSPublicKey): PSSignature {
+    return new PSSignature(psUnblindSignature(this.value, indexedBlindings, pk.value));
+  }
+
+  /**
+   * Generate a request for a blind signature
+   * @param messagesToBlind - messages the requester wants to hide from the signer. The key of the map is the index of the
+   * message as per the params.
+   * @param blindings - If not provided, a random blindings will be generated for each message
+   * @param params
+   * @param h
+   * @param blinding
+   * @param revealedMessages - Any messages that the requester wishes to inform the signer about. This is for informational
+   * purpose only and has no cryptographic use.
+   */
+  static generateRequest(
```

Comment:
The order of args is a bit different from other 2 schemes. All blindings should follow params or come before them. Also we sometime say `revealedMessages` and sometimes `unblindedMessages`. Let's use a single term, prefer revealed as its less technical.